department-of-veterans-affairs/caseflow
# Refactored to fix invalid year bug.

Resolves [Incorrect Receipt Date Filtering UI Error](https://jira.devops.va.gov/browse/APPEALS-30049)

# Description
Previously, when users entered the date manually, the receipt filter would error out because it was translating the first character entered as a full valid year. These changes make it so that validation occurs after the user hits the filter button, which kicks off `executeRecieptFilter()`.

## Acceptance Criteria

## Testing Plan

1. Login to caseflow as a user who can access reader (BVAAABSHIRE)
2. Navigate to reader
3. filter the date between
4. enter the from date (any date not in the future, and occurring before the to date, and also not on the same date as the to date)
5. enter the to date manually by typing in the date field, (09-09-2023)
6. The user no longer encounters the error and the errors will be displayed if there are any after the filter button is clicked.

# Frontend
## User Facing Changes
Screenshots were basically impossible to produce for this issue and have been omitted.
